### PR TITLE
Main: Work around unserializable object exception from CJS in formatLogArgument, improve indentation and JSON serializable object visibility

### PR DESCRIPTION
**Pull request by jaszhix**
How to reproduce: run `global.log` on an AppletSettings instance. 

It turns out there are some native objects [CJS can't stringify](https://github.com/linuxmint/cjs/blob/f7638496ea1bec4c6774e6065cb3b2c38b30a7bf/cjs/context.cpp#L138), so a try-catch block is used to short circuit the recursion in this scenario.

### CODE REVIEW

**Comment by clefebvre** [comment]
I get no logs in LG with this PR...

I really don't like some of the code constucts in there... (not just your PR but the preexisting code..).. for instance

```
let space = Array.apply(null, Array(recursion))
        .map(String.prototype.valueOf, '    ')
        .join('');
```

I'd rather see lengthy loops and if statements than mappings like these.

**Comment by jaszhix** [comment]
I'm seeing all the normal Cinnamon initialization messages in glass.log. When I insert `global.log('actor', this.actor)` in an applet's _init method, I see...

```
actor: {
_applet: {
    actor: [0x55952e453740 StBoxLayout.applet-box ("Initializing")],
    _allowedLayout: "'both'",
    _signalConnections: [{"id":1,"name":"orientation-changed","disconnected":false},{"id":2,"name":"orientation-changed","disconnected":false}],
    _nextConnectionId: 3,
    _applet_tooltip: [object Object],
    _menuManager: [object Object],
    _applet_context_menu: [object Object delegate for 0x55952e443390 StBin.menu:last-child],
    _order: 0,
    _newOrder: "null",
    _panelLocation: "null",
    _newPanelId: "null",
    _newPanelLocation: "null",
    _applet_enabled: true,
    _orientation: 2,
    _panelHeight: 27,
    instance_id: "'171'",
    _uuid: "null",
    _hook: "null",
    _meta: "null",
    _dragging: false,
    _draggable: [object Draggable],
    _scaleMode: false,
    _applet_tooltip_text: "''",
    context_menu_item_remove: "null",
    context_menu_separator: "null",
    _panelEditModeChangedId: 20942,
    _applet_icon_box: {},
    _applet_label: {},
    _layoutBin: {},
    show_label_in_vertical_panels: true,
},
_delegate: {
    actor: [0x55952e453740 StBoxLayout.applet-box ("Initializing")],
    _allowedLayout: "'both'",
    _signalConnections: [{"id":1,"name":"orientation-changed","disconnected":false},{"id":2,"name":"orientation-changed","disconnected":false}],
    _nextConnectionId: 3,
    _applet_tooltip: [object Object],
    _menuManager: [object Object],
    _applet_context_menu: [object Object delegate for 0x55952e443390 StBin.menu:last-child],
    _order: 0,
    _newOrder: "null",
    _panelLocation: "null",
    _newPanelId: "null",
    _newPanelLocation: "null",
    _applet_enabled: true,
    _orientation: 2,
    _panelHeight: 27,
    instance_id: "'171'",
    _uuid: "null",
    _hook: "null",
    _meta: "null",
    _dragging: false,
    _draggable: [object Draggable],
    _scaleMode: false,
    _applet_tooltip_text: "''",
    context_menu_item_remove: "null",
    context_menu_separator: "null",
    _panelEditModeChangedId: 20942,
    _applet_icon_box: {},
    _applet_label: {},
    _layoutBin: {},
    show_label_in_vertical_panels: true,
},
}
```
The chained methods you pointed out are creating indentation spaces for the object formatting, but the bulk of that code is never reached (evaluated, loaded into memory), unless an object or an array is passed to the function. 

`Array.apply(null, Array(recursion))`

Creates a new array, with a length of `recursion` - representing the depth level of an object structure, so its used for the indentation count as well.

`.map(String.prototype.valueOf, '    ')`

This replaces the empty array elements with a string with four spaces.

`.join('')`

This converts the array to a string.

This function used to be easier to read. I formatted this PR for ES6 (mozjs38) after being told Cinnamon 3.6 would be expected to be packaged with CJS 3.4. I guess minds were changed in light of the CJS segfaults, which I attempted to help out with.

I can break those pieces of code into a pair of for loops, and then maybe that might be moved to its own function for creating indentations for the object logging? I try to walk a healthy line between repeating myself too much, and abstracting too much. In JS, leveraging chaining makes these operations more compact, and in my opinion, not too hard to follow once broken down to individual lines.

**Comment by jaszhix** [comment]
Okay, logging should be fixed, and fixed the weird indentation seen in the example I pasted earlier. I like the for loop approach better anyway, its faster than `map`.
